src/ucp/wireup/wireup.c

```diff
@@ -125,11 +126,11 @@ ucs_status_t ucp_wireup_msg_progress(uct_pending_req_t *self)
 
     packed_len = uct_ep_am_bcopy(ep->uct_eps[req->send.lane], UCP_AM_ID_WIREUP,
                                  ucp_wireup_msg_pack, req, am_flags);
-    UCP_AM_BCOPY_HANDLE_STATUS(0, (ucs_status_t)packed_len);
-    if (ucs_unlikely(packed_len < 0)) {
-        ucs_assert((ucs_status_t)packed_len != UCS_ERR_NO_RESOURCE);
-        ucs_error("failed to send wireup: %s",
-                  ucs_status_string((ucs_status_t)packed_len));
+    status     = (packed_len > 0) ? UCS_OK : (ucs_status_t)packed_len;
+    UCP_AM_BCOPY_HANDLE_STATUS(0, status);
+    if (ucs_unlikely(status != UCS_OK)) {
+        ucs_assert(status != UCS_ERR_NO_RESOURCE);
+        ucs_error("failed to send wireup: %s", ucs_status_string(status));
```

Comment:
i think we need to make most of this part of UCP_AM_BCOPY_HANDLE_STATUS
so the status of any bcopy operation will be handled correctly

Reply:
> i think we need to make most of this part of UCP_AM_BCOPY_HANDLE_STATUS

but then the macro will be useless for `ucp_do_am_bcopy_single`/`ucp_do_am_bcopy_multi`

Reply:
the best solution is to re-use `ucp_do_am_bcopy_single()` or even `ucp_do_am_single()` to send WIREUP MSG as well.
we can implement common part for `ucp_do_am_bcopy_single()` which won't get sending lane and use it in `ucp_do_am_bcopy_single()` and for sending WIREUP MSG, i.e:
```
ucs_status_t ucp_do_am_bcopy_single(uct_pending_req_t *self, uint8_t am_id, uct_pack_callback_t pack_cb)
{
    ucp_request_t *req   = ucs_container_of(self, ucp_request_t, send.uct);
    req->send.lane = ucp_ep_get_am_lane(req->send.ep);
    return ucp_do_am_bcopy_single_common(req, am_id, pack_cb, req, 0);
}
```
and
```
ucp_wireup_msg_progress
{
   ucp_request_t *req   = ucs_container_of(self, ucp_request_t, send.uct);
    ...
    req->send.lane = ucp_ep_get_wireup_msg_lane(req->send.ep);
    return ucp_do_am_bcopy_single_common(req, UCP_AM_ID_WIREUP, ucp_wireup_msg_pack, req, am_flags);
}
```